docs/proposed/psa-driver-interface.md

```diff
@@ -607,7 +616,11 @@ Note that a solution also has to work for transparent keys, and when importing a
 
 #### Opaque driver persistent state
 
-Should the driver be able to update it at any time?
+The driver is allowed to update the state at any time. Is this ok?
+
+An example use case for updating the persistent state at arbitrary times is to renew a key that is used to encrypt communications between the application processor and the secure element.
+
+`psa_crypto_driver_get_persistent_state` does not identify the calling driver, so the driver needs to remember which driver it's calling. This may require a thread-local variable in a multithreaded core. Is this ok?
```

Comment:
I think this'll cause a lot of boilerplate in the code that is calling driver entry points. If we can contain the boilerplate to the autogen'ed wrapper file that could be fine, but if this particularity is going to leak into psa_crypto.c it's another story.

Reply:
That's a good point. I liked this being automatic to reduce the risk of mistakes. But maybe `psa_crypto_driver_get_persistent_state` should take a `lifetime` argument, and _optionally_ the core could refuse if the lifetime is wrong?

Reply:
Then you could end up with rogue drivers trying to grab each other's state... Not sure whether that's a concern or not.

Reply:
There's no memory isolation between drivers. At least not on typical implementations. A high-security implementation could isolate drivers (and make sure they have sufficiently restrictive DMA permission so that they can't access arbitrary memory via the crypto hardware!), and then it had better check the lifetime.

Reply:
On the other hand, having to pass the location will create similar boilerplate in drivers. So I prefer to leave it this way for now and wait to see how it goes in practice.